src/ac_copilot_trainer/modules/lap_archive.lua

```
@@ -0,0 +1,340 @@
+--- Returns (filesKept, mbUsed, filesDeleted).
+---@param capMB number
+---@return integer, number, integer
+function M.rotate(capMB)
+  capMB = tonumber(capMB) or 500
+  if capMB <= 0 then capMB = 500 end
+  local capBytes = capMB * 1024 * 1024
+  local dir = lapArchiveDir()
+  -- io.scanDir is a CSP API; fall back to noop if unavailable
+  local files = {}
+  local okScan = pcall(function()
+    if io and type(io.scanDir) == "function" then
```

Luacheck (1.2.0)

[warning] accessing undefined field 'scanDir' of global 'io'

(W143)